ASC updates (#2077)
ppy/osu-wiki: wiki/Article_Styling_Criteria/en.md

*For more information about HTML entities, see: [Entity - Glossary](https://developer.mozilla.org/en-US/docs/Glossary/Entity "MDN")*

> HTML character entities should not be used. If used however, prefer the character over the entity. For example:

Author: > I don't think the [...]

Huh, didn't think that they were part of HTML despite having it in the name. Fwiw, I copy it from Google or use amp-what.

> We can add special case sections [...]

Oh yeah, any thoughts on the wording for this section? I'm going to go along the lines of: `HTML comments may be used to annotate text. Keep usage limited.`

This can be as small as changing the cursor or as big as redoing the appearances of all the game modes and interface.
```

Both will look like this:

![](img/paragraphs.png)

#### Soft breaks

<!-- ppy ruled soft breaks are not allowed. from:ppy#0001 during:2017-07-12 in:osu-wiki paragraph -->

If an empty line is not present between two lines (also known as a soft break, henceforth), the two lines will merge into a single paragraph. Soft breaks are not allowed in the osu! wiki. For example:

```md
Skinning allows anyone to change the way osu! looks and feels.
This can be as small as changing the cursor or as big as redoing the appearances of all the game modes and interface.
```

Which is the same as:

```md
Skinning allows anyone to change the way osu! looks and feels. This can be as small as changing the cursor or as big as redoing the appearances of all the game modes and interface.
```

And both will look like this:

> ![](img/soft-breaks.png "Although the text is in two sepatate lines, it is parsed as a single paragraph.")

Reviewer: ```suggestion
![](img/soft-breaks.png "Although the text is in two separate lines, it is parsed as a single paragraph.")
```

### Headings

Do not use headings to style or format text and limit to a heading level of 5. Use the [ATX (hash) style](https://github.github.com/gfm/#atx-headings). Each heading should be preceded and followed by an empty line.

#### Titles

*Notice: Article titles must not be escaped, as they are parsed as plain text, not Markdown.*

> The first heading in all articles must be a level 1 heading, being the article's title. All headings that follow the title heading must be [section headings](#sections). Article titles must use title case and must not contain formatting, links, nor images.

Reviewer: ```suggestion
The first heading in all articles must be a level 1 heading, being the article's title. All headings that follow the title heading must be [section headings](#sections). Article titles must use title case and must not contain formatting, links, or images.
```

This level 1 heading must be placed at the start of the article unless reference links or metadata placed there. If this is the case, place the level 1 heading underneath it.

The title of an article must match the folder name it is located in (spaces can replace underscores (`_`) where appropriate). The exception to this rule are tournament pages, which may use abbreviations, acronyms, or initialisms.

---

For English article titles, if the article's title needs to be renamed, the folder name it is in must also be renamed to match it.

#### Sections

> All section headings must use levels 2 to 5. Section headings must use sentence case and must not contain formatting nor links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the [title heading](#titles) must be a level 2 heading. Unlike titles, section headings can have small image icons.

Reviewer: ```suggestion
All section headings must use levels 2 to 5. Section headings must use sentence case and must not contain formatting or links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the [title heading](#titles) must be a level 2 heading. Unlike titles, section headings can have small image icons.
```

#### Titles

*Notice: Article titles must not be escaped, as they are parsed as raw text, not Markdown.*

All articles must have one level 1 heading, being the article's title. Article titles must use title case and never contain formatting, links, nor images. The level 1 heading must be placed at the start of the article unless reference links or metadata placed there. If this is the case, place the level 1 heading underneath it.

The title of an article must match the folder name it is located in (spaces can replace underscores (`_`) where appropriate). The exception to this rule are tournament pages, which may use abbreviations, acronyms, or initialisms.

---

For English article titles, if the article's title needs to be renamed, the folder name it is in must also be renamed to match it.

#### Sections

> All section headings must use levels 2 to 5, never level 6. Section headings must use sentence case and must not contain formatting nor links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the title heading must be a level 2 heading. Unlike titles, section headings can have small image icons.

Reviewer: Do not need to mention as there is a range stated.
```suggestion
All section headings must use levels 2 to 5. Section headings must use sentence case and must not contain formatting nor links. In addition to this, section headings must not skip a heading level (e.g. do not go from a level 2 heading to a level 4 heading). The section heading after the title heading must be a level 2 heading. Unlike titles, section headings can have small image icons.
```

> In the osu! wiki, heading levels 4 to 6 do not appear in the table of contents and are not given the id attribute (these cannot be linked directly).

Reviewer: > All section headings must use levels 2 to 5.
```suggestion
In the osu! wiki, heading levels 4 and 5 do not appear in the table of contents and are not given the id attribute (these cannot be linked directly).
```

Author: I thought it should still be mentioned even if it is not to be used, but I suppose it might lead to some confusion.

#### Keyboard keys

When doing this, use capital letters for single characters and title case for modifiers. Denoting the letter itself, not the keyboard key, should not be in code (it should use quotes). For example:

```md

> pippi is spelt with a lower case "P" like peppy.

Reviewer: ```suggestion
pippi is spelt with a lowercase "P" like peppy.
```

Do:     [osu! on the web](https://osu.ppy.sh/home)
```

#### Internal links

> *Note: Internal links refers to links that stay inside the `https://osu.ppy.sh/` domain.*

Reviewer: ```suggestion
*Note: Internal links refer to links that stay inside the `https://osu.ppy.sh/` domain.*
```

```

*Note: It is not possible to float an image or have text wrap around it. If the image is placed inline with the text, the image will be displayed inline with the text.*

#### Alternate and title text

All images should have alternate text. This will ensure that something will be displayed even if the image fails to load. Title text is optional and could be the same as the alternate text if needed.

#### Captions

Images are automatically given captions if they fulfill these conditions:

1. The image is by itself.
2. The image is not inside a heading.
3. The image has title text.

Captions are assumed via the title text. Image captions must be plain text without the. Images with captions are also centred with the image on the website.

#### Max image width

The osu! wiki gives every image a max-width, being the width of the article body (680 pixels). When an image's width is more than 680 pixels, consider downscaling it.

#### Annotating images

When annotating images, use *Exo 2 light*. For Chinese, Korean, Japanese characters, use *Microsoft YaHei*.

Try to limit writing text in images, as it is difficult for translators (and other editors) to edit them.

> 

Reviewer: Any reason removing this? It explains a proper format.

This can surely be used when taking pictures of the osu! client, which has locales/translations.

##### Translating annotated images

> When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:	

Reviewer: ```suggestion
When translating annotated images, the localised image version must be placed in the same directory as the original version (i.e. the English version). The filename of a localised image version must start with the original version's name, followed by a hyphen, followed by the locale name (in caps). For example:
```

> - `hardrock_mod_vs_easy_mod.jpg` (i.e. in English)	

Reviewer: ```suggestion
- `hardrock_mod_vs_easy_mod.jpg` (i.e. in English)
```

> - `hardrock_mod_vs_easy_mod-DE.jpg` (i.e. in German)	

Reviewer: ```suggestion
- `hardrock_mod_vs_easy_mod-DE.jpg` (i.e. in German)
```

![](/wiki/shared/flag/xx.gif)
```

Where `xx` is the [ISO 3166-2](https://en.wikipedia.org/wiki/ISO_3166-1_alpha-2 "Wikipedia") two-lettered country code for the flag.

> Adding the full country names or codes in the alt text is optional. Adding full country names in title text is optional.

Reviewer: ```suggestion
Adding the full country names or codes in the alternate text is optional. Adding full country names in title text is optional.
```

*Notice: All image file extensions must use lowercase letters, otherwise they will fail to load!*

> Image file names must not use spaces (use underscores (`_`) or hyphens (`-`) instead). When naming an image, the file name should be meaningful or descriptive but short. Never use timestamps or long hashes for the file name.

Reviewer: Can you prefer a character for the spacing to resolve #1996.

Author: Alright, we'll go with hyphens, for [this reason](https://github.com/ppy/osu-wiki/issues/1996#issuecomment-457861162)?

Reviewer: Also the OP's reason, yeah.

### osu!

> The name of the game, osu!, must not capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:

Reviewer: ```suggestion
The name of the game, osu!, must not be capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:
```

### osu!

> The name of the game, osu!, must not be capitalised nor italicised. The osu! official branding must not use any spaces. Examples include:

Reviewer: ```suggestion
The name of the game, osu!, must not be capitalised or italicised. The osu! official branding must not use any spaces. Examples include:
```

#### Proper nouns

Proper nouns must be first-letter capitalised. For example:

```
Dean Herbert (also known as peppy) created osu!.
```

#### Trademarks

> *For rules regarding osu!, see [osu!](#osu!).*

Reviewer: ```suggestion
*For rules regarding osu!, see: [osu!](#osu!)*
```

#### Proper nouns

Proper nouns must be first-letter capitalised. For example:

```
Dean Herbert (also known as peppy) created osu!.
```

#### Trademarks

*For rules regarding osu!, see: [osu!](#osu!)*

The following trademarks must be spelt as follows (letter casing must match):

- `Discord`
- `Facebook`
- `GitHub`
- `Google`
- `Reddit`
- `Skype`
- `Twitch`
- `Twitter`
- `YouTube`

> Do not use the trademark nor registered trademark symbols with trademarks.

Reviewer: ```suggestion
Do not use the trademark or registered trademark symbols with trademarks.
```

- `osu!catch`
- `osu!mania`

*Notice: `osu!standard` is used to maintain consistency when referring to the game mode. Folder names, however, must use `osu!` even if it is referring to the game mode.*

Referring to the old game mode names (i.e. `Catch the Beat`, `Taiko`, and `Mania`) can be done if discussing about said game mode's previous name.

Articles that are more official, e.g. [Ranking Criteria](/wiki/Ranking_Criteria), may use `osu!` instead of `osu!standard`.

### Terminology

These words are spelt as follows (spacing must match):

- `approach circle`
- `game mode` (or `mode`)
- `game mod` (or `mod`)
- `gameplay`
- `hit burst`
- `hit circle`
- `hit sound`
- `in-game`

> - `play style`

Reviewer: Either use the suggestion below (idk why it does this on non-diffs) or commit another way.
```suggestion
- `playstyle`
```

Author: We are going with "playstyle" now?

Author: Never mind, looked back at the Discord again.

- `slider tick`
- `slider ball`
- `slider path`

---

- Use `beatmap` instead of `map`.
- Use `creator` instead of `beatmapper` or `mapper`.

> - Use `created` instead of `beatmapped` or `mapped`.

Reviewer: Still curious as to why we don't follow `osu-web`'s terminology with `mapped`.

Author: It was to go with the term above "creator", will change it to "mapped" instead.

---

Avoid pluralising abbreviations, acronyms, and/or initialisms; the sentence should be rephrased to avoid this. For example:

```
Avoid: BNs can nominate your beatmapset.

Do:    Members of the Beatmap Nominators can nominate your beatmapset.
```

### Date formatting

> The osu! wiki will use the [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) format, which is `YYYY-MM-DD`. For example:

Reviewer: ```suggestion
The osu! wiki uses the [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) format, which is `YYYY-MM-DD`. For example:
```
